lib/YourlsProxy.php

```diff
@@ -76,56 +64,25 @@ public function __construct(Configuration $conf, $link)
                 )
             )
         );
-        try {
-            $data = Json::decode($data);
-        } catch (Exception $e) {
-            $this->_error = 'Error calling YOURLS. Probably a configuration issue, like wrong or missing "apiurl" or "signature".';
-            error_log('Error calling YOURLS: ' . $e->getMessage());
-            return;
-        }
+    }
 
+    /**
+     * Extracts the short URL from the YOURLS API response.
+     *
+     * @access protected
+     * @param array $data
+     * @return ?string
+     */
+    protected function _extractShortUrl(array $data): ?string
+    {
         if (
             !is_null($data) &&
```

Comment:
Same here, as [Scrutinizer points out](https://scrutinizer-ci.com/g/PrivateBin/PrivateBin/inspections/5259d197-a9b9-467d-b00e-1925aec8c386/issues/files/lib/YourlsProxy.php?status=new&orderField=path&order=asc&honorSelectedPaths=0), because we now enforce $data to be an array in the method signature, this check can no longer fail.
```suggestion
```